GUIDE.md

```diff
@@ -1,28 +1,160 @@
-# Server Endpoint
+# Guide
 
+ - The guiding focus to how to write a client to connect to a Tokyo server, receive information events and create actions to control your ship
+
+## 1. Connection information
+
+`ws://${host}/socket?key={key}&name{name}`
+
+| Parameters | Description |
+|--|--|
+| {host} | Domain or IP of server want to connect |
+| {key} | User's identity is unique and used to distinguish bots |
+| {name} | The display name will be shown on UI |
+
+When the WebSocket connection is established successfully, your bot is registered, displayed on web UI and ready to use.
+
+## 2. Events
+
+From WebSocket, the server consecutively sends events to the client every tick with the following structure.
+Event message structure contains all world info of the game, including players, bullets, dead person, etc.
+
+### 2.1. Event general structure
+
+```json
+{
+  "e": "state",
+  "data": {
+    "bounds": [1200.0, 800.0],
+    "players": [
+      {player},{player}
+    ],
+    "bullets": [
+      {bullet},{bullet}
+    ],
+    "dead": [
+      "respawn":{
+		    "secs_since_epoch":1631299999,
+	      "nanos_since_epoch":940876053
+       },
+       "player": {player}
+    ],
+    "scoreboard":{"0":100,"1":90,"2":80}
+  }
+}
 ```
-$HOST/socket
+
+| Fields | Description |
+|--|--|
+| e | Event information. The event from the server is always "state" |
+| data | Detail data of event "e" |
+| bounds | Boundary of the game, players spawn and navigate their ship in boundary from position [0,0] to this max size boundary. It's an array with two values, width and height |
+| players | List of players/ships in the game currently. Detail of the player object will be described in the next sections |
+| bullets | List of bullets that's fired by ships in the game currently. Detail of bullet object will be described in the next sections |
+| dead | List of dead users and the respawn periods. Information of player is a structure with "players" |
+| scoreboard | Top user score with format "player_id: score" |
+
+### 2.2. Player structure
+
+```json
+{
+   "id":0,
+   "angle":9.350119,
+   "throttle":1.0,
+   "x":579.5356,
+   "y":118.02286
+},
 ```
 
+| Fields | Description |
+|--|--|
+| id | Player/ship's identify (ID) |
+| angle | Angle of the ship is heading. Radian value's between [0, 2π] |
+| throttle | Throttle or speed of the ship. 0 = no speed, 1 = max speed |
+| x, y | Ship's position |
+
+### 2.3. Bullet structure
 
+```json
+{
+   "id":564,
+   "player_id":1,
+   "angle":8.630102,
+   "x":1013.78644,
+   "y":312.22202
+}
 ```
+
+| Fields | Description |
+|--|--|
+| id | Bullet's identify (ID) |
+| player_id | Identify the ship that fires this bullet |
+| angle | Angle of the bullet is heading. It will move forward |
```

Comment:
`It will move forward` is kinda redundant I think.

Reply:
To avoid everyone thinks the ship move backwards, just kidding